### PR TITLE
Javascript atom - improvement

### PR DESCRIPTION
**Pull request by adi8i**
#577 improvement

### CODE REVIEW

**Comment by w00fz** [comment]
Nice!

**Comment by w00fz** [comment]
It just needs to be reviewed, thanks @adi8i, good work!

File: engines/common/nucleus/particles/assets.html.twig [hunk 1]
```diff
@@ -25,7 +26,36 @@
                 {% endfor %}
             {% endfor %}
         {% endif %}
+        
+        <script src="{{ url(script.location) }}" type="text/javascript"{{ attr_extra|raw }}></script>
```
Comment (mahagr) [inline engines/common/nucleus/particles/assets.html.twig]: This line should be in {% else %} of the next if.

File: engines/common/nucleus/particles/assets.html.twig [hunk 2]
```diff
@@ -25,7 +26,36 @@
...
+        {% set attr_extra = '' %}
+        {% if script.extra %}
+            {% for attributes in script.extra %}
+                {% for key, value in attributes %}
+                    {% set attr_extra = attr_extra ~ ' ' ~ key|e ~ '="' ~ value|e('html_attr') ~ '"' %}
+                {% endfor %}
+            {% endfor %}
+        {% endif %}
+        
         <script src="{{ url(script.location) }}" type="text/javascript"{{ attr_extra|raw }}></script>
```
Comment (mahagr) [inline engines/common/nucleus/particles/assets.html.twig]: This line should be in {% else %} of the next if.

**Comment by mahagr** [comment]
Actually my comments isn't the full story. I'm just trying to tell that if you have just inline js set and no file, it's still going to add tag for the file.

So the question is how this should work?

File: engines/common/nucleus/particles/assets.html.twig (hunk 2 above)
Comment (alisamii) [inline engines/common/nucleus/particles/assets.html.twig]: Does the code, as is, just insert a link to an external script file at the end of the body tag? Or does it inline the script?

Is there any benefit to inlining the script over just linking to the script other than reducing http request(s)?

Best regards,

Ali

**Comment by w00fz** [comment]
Thank you @adi8i this was an awesome PR!

**Pull request merged**